include/banana/utils/agent.hpp

```diff
@@ -147,15 +147,16 @@ struct make_callback : Agent {
     template <class Traits, class F, class R = typename Traits::response_type>
     void request(std::string body, F&& callback) {
         struct callback_handler : async_handler {
-            F callback;
```

Comment:
No need for `std::function` - it's just one more level of indirection with possible memory allocation. Could you replace it with `std::remove_const_t<std::remove_reference_t<F>> callback;`? Or, simpler, we can accept `F callback` (not `F&&`) everywhere in APIs and intermediate functions, move it and store as a simple `F` (like we do with `args`).

Reply:
Yes, my bad! Adding a constructor is sufficient to allow passing lambda with captures. I will remove std::function. By the way, should I also remove `#include <functional>` from this file? It seems unused.

Reply:
Yes, sure